west.yml: add crypto driver for TL721X A1

pulin1103 opened the pull request:
- enable CI crypto demo.
- update TL721X pke header file.
- rename TLX eccp_curve_t structure.
- add pke_x25519_point_mul dummy func to compile.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | pull/116/head | https://github.com/telink-semi/hal_telink/commit/c09858d598f868c830b21337d0cdf10aecb129cf (tl721x_draft) | [telink-semi/hal_telink@pull/116/head..c09858d5](https://github.com/telink-semi/hal_telink/compare/pull/116/head..c09858d598f868c830b21337d0cdf10aecb129cf) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

Commit:
west.yml: add crypto driver for TL721X A1

- enable CI crypto demo.
- update TL721X pke header file.
- rename TLX eccp_curve_t structure.
- add pke_x25519_point_mul dummy func to compile.

Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>
